drivers/gpio/gpio_andes_atcgpio100.c

```diff
@@ -0,0 +1,370 @@
+
+	k_spin_unlock(&data->lock, key);
+
+	return 0;
+}
+
+static int gpio_atcgpio100_pin_interrupt_configure(
+					const struct device *port,
+					gpio_pin_t pin,
+					enum gpio_int_mode mode,
+					enum gpio_int_trig trig)
+{
+	struct gpio_atcgpio100_data * const data = DEV_DATA(port);
+	uint32_t port_value, int_mode, imr_idx, ch_idx;
+	k_spinlock_key_t key;
+
+	switch (mode | trig) {
+	case GPIO_INT_EDGE_BOTH:
+		int_mode = INT_DUAL_EDGE;
+		break;
+	case GPIO_INT_EDGE_RISING:
+		int_mode = INT_POSITIVE_EDGE;
+		break;
+	case GPIO_INT_EDGE_FALLING:
+		int_mode = INT_NEGATIVE_EDGE;
+		break;
+	case GPIO_INT_LEVEL_LOW:
+		int_mode = INT_LOW_LEVEL;
+		break;
+	case GPIO_INT_LEVEL_HIGH:
+		int_mode = INT_HIGH_LEVEL;
+		break;
+	default:
+		int_mode = INT_NO_OPERATION;
+		break;
+	}
+
+	imr_idx = (pin >> 3);
+	ch_idx = (pin & 0x7);
+
+	key = k_spin_lock(&data->lock);
+
+	if (int_mode == INT_NO_OPERATION) {
+		/* Disable interrupt of pin */
+		port_value = INWORD(GPIO_INTE(port));
+		OUTWORD(GPIO_INTE(port), port_value & ~(1 << pin));
+
+		/* Clear the remain pending interrupt */
+		port_value = INWORD(GPIO_ISTA(port));
+		OUTWORD(GPIO_ISTA(port), port_value);
+	} else {
+		/* Set interrupt mode of pin */
+		port_value = INWORD(GPIO_IMD(port, imr_idx));
+		SET_GPIO_INT_MODE(port_value, int_mode, ch_idx);
+		OUTWORD(GPIO_IMD(port, imr_idx), port_value);
+
+		/* Enable interrupt of pin */
+		port_value = INWORD(GPIO_INTE(port));
+		OUTWORD(GPIO_INTE(port), port_value | (1 << pin));
+	}
+
+	k_spin_unlock(&data->lock, key);
+
+	return 0;
+}
+
+static int gpio_atcgpio100_manage_callback(const struct device *port,
+					struct gpio_callback *callback,
+					bool set)
+{
+
+	struct gpio_atcgpio100_data * const data = DEV_DATA(port);
+
+	return gpio_manage_callback(&data->cb, callback, set);
+}
+
+static void gpio_atcgpio100_irq_handler(const struct device *port)
+{
+	struct gpio_atcgpio100_data * const data = DEV_DATA(port);
+	uint32_t port_value;
+
+	port_value = INWORD(GPIO_ISTA(port));
+	OUTWORD(GPIO_ISTA(port), port_value);
+
+	gpio_fire_callbacks(&data->cb, port, port_value);
+
+}
+
+static const struct gpio_driver_api gpio_atcgpio100_api = {
+	.pin_configure           = gpio_atcgpio100_config,
+	.port_get_raw            = gpio_atcgpio100_port_get_raw,
+	.port_set_masked_raw     = gpio_atcgpio100_set_masked_raw,
+	.port_set_bits_raw       = gpio_atcgpio100_set_bits_raw,
+	.port_clear_bits_raw     = gpio_atcgpio100_clear_bits_raw,
+	.port_toggle_bits        = gpio_atcgpio100_toggle_bits,
+	.pin_interrupt_configure = gpio_atcgpio100_pin_interrupt_configure,
+	.manage_callback         = gpio_atcgpio100_manage_callback
+};
+
+static int gpio_atcgpio100_init(const struct device *port)
+{
+	const struct gpio_atcgpio100_config * const dev_cfg = DEV_CFG(port);
+
+	/* Disable all interrupts */
+	OUTWORD(GPIO_INTE(port), 0x00000000);
+
+	/* Write 1 to clear interrupt status */
+	OUTWORD(GPIO_ISTA(port), 0xFFFFFFFF);
```

Comment:
You could use here (and anywhere else where you use bitmask) the `BIT_MASK` macro from `include/sys/util_macro.h`.

Reply:
Rebased to BIT / BIT_MASK macro in Zephyr.

But there is a 1 problem. `BIT_MASK(32)` will have integer overflow in `BIT(32)` and get compiler warning (-Wshift-count-overflow).
I have used `BIT64_MASK` and converted to `uint32_t` to workaround it. 
(Compiler could optimize it, but it doesn't seem to be clear?)

Reply:
I think using `(uint32_t) BIT64_MASK(32)` is OK in this PR, so I will resolve this conversation first. 
If having any problem, you can reopen it.